src/FunctionBlocks/LogicFunctionBlock.cpp

```diff
@@ -146,9 +147,10 @@ void LogicFunctionBlock::handleKo(GroupObject& ko)
             }
             auto& ko = getKo(i);
 
-            auto inputValue = (bool)ko.value(DPT_Switch);
+            bool inputValue = ko.value(DPT_Switch);
             if (inputKo == 2)
                 inputValue = !inputValue;
+            
             if (_type == LogicFunctionBlockType::LogicAND)
```

Comment:
Hier würde ich nun auch auf ein switch statt der if elseif Variante gehen. Alternative überlegung wäre sonst, dass LogicFunctionBlock abstract wird, und wir die Typespezifischen Dinge in den Ableitungen machen.

Reply:
Switch wird in Kombination mit der aktuellen Nutzung des `break` (mit dem Du die Schleife verlässt wenn das Ergebnis fest steht) unschön. Sonst hätte ich das auch schon direkt mit umgestellt ;-)

Reply:
Man könnte statt dem break den Schleifenzähler auf einen großen Wert stellen, aber ob das schöner ist weiß ich auch nicht.

Reply:
Wahrscheinlich wäre am Besten mit Ableitungen zu Arbeiten und dann eben die Schleife mehrfach implementiert zu haben.

Reply:
> Man könnte statt dem break den Schleifenzähler auf einen großen Wert stellen, aber ob das schöner ist weiß ich auch nicht.

Das wird vom Kontrollfluss dann eher unübersichtlicher.

> […] mit Ableitungen zu Arbeiten und dann eben die Schleife mehrfach implementiert zu haben.

Mehrfach Implementieren der Schleife würde ich vermeiden wollen. Dafür steckt zu viel in der Vorverarbeitung drin. Bei Vervielfältigung der Implementierung läuft das absehbar irgendwann auseinander. Template- oder Strategy-Pattern (Idee: `bool canBreak = aggregate(value)`) wären ggf. eine Option, würde aber den Code-Umfang/Dateianzahl an dieser Stelle auch deutlich erhöhen. Eine Verbesserung dadurch wäre daher aus meiner Sicht fragwürdig.

Alternativ könnten wir auch ganz auf die Fallunterscheidung (und Optimierung für AND/OR) verzichten und nur die Eingänge==1 zählen:
```
AND  :<=> (anzahl1 == anzahlGenutzt)
OR   :<=> (anzahl1 > 0)
COUNT :=   anzahl1
```